### DISCUSSION TITLE
Cypress cannot visit a website that restricts IP addresses

### DISCUSSION THREAD

**Discussion by IWTBCAdeveloper** [cypress-io/cypress#27058] [Questions and Help]
Hi all, I need to use Cypress to visit a website that restricts IP addresses, so how should I add a whitelist IP address of Cypress, please help

**Comment by portyu9** [comment]
Hi @IWTBCAdeveloper 

There isn't a single public **"Cypress IP address"** that you should whitelist.

Your application sees the network egress IP of the machine/network where the Cypress browser is actually running.

So:

If Cypress runs on your laptop, whitelist your network/VPN/NAT's public egress IP.

If Cypress runs in CI, whitelist the CI runner's outbound IP. If your CI provider uses dynamic IP addresses, you will need either the provider's documented ranges or, preferably, infrastructure with a stable outbound address such as a static NAT, VPN, or self-hosted runner.

Recording the run to **Cypress Cloud does not cause the requests to your application to originate from Cypress Cloud**. Cypress Cloud complements your CI system; the actual Cypress tests still run on your CI runner.

So the IP allowlist belongs to your network/CI infrastructure rather than Cypress configuration.

Please mark this answer as accepted if it helped, thank you.
